package/MDAnalysis/coordinates/TRC.py

```diff
@@ -0,0 +1,399 @@
+                    self.periodic = False
+
+                elif (ntb_setting in [-1, 1]):
+                    tmp_a, tmp_b, tmp_c = f.readline().split()
+                    tmp_alpha, tmp_beta, tmp_gamma = f.readline().split()
+                    frameDat["dimensions"] = [float(tmp_a),
+                                              float(tmp_b),
+                                              float(tmp_c),
+                                              float(tmp_alpha),
+                                              float(tmp_beta),
+                                              float(tmp_gamma)]
+                    self.periodic = True
+
+                    line3 = f.readline().split()
+                    line4 = f.readline().split()
+                    for v in (line3 + line4):
+                        if (float(v) != 0.0):
+                            raise NotImplementedError("This reader \
+                                                      supports neither \
+                                                      triclinic and/or \
+                                                      (yawed,pitched, \
+                                                      rolled) boxes!")
+                else:
+                    raise NotImplementedError("This reader does only support\
+                                               vacuum and rectangular boxes!")
+                break
+
+            elif any(non_supp_bn in line for
+                     non_supp_bn in self.NOT_SUPPORTED_BLOCKNAMES):
+                for non_supp_bn in self.NOT_SUPPORTED_BLOCKNAMES:
+                    if (non_supp_bn == line.strip()):
+                        warnings.warn(non_supp_bn + " block is not supported!",
+                                      UserWarning)
+
+        return frameDat
+
+    def _read_frame(self, i):
+        """read frame i"""
+        self._frame = i - 1
+
+        # Move position in file just (-2 byte) before the start of the block
+        self.trcfile.seek(self.traj_properties["l_blockstart_offset"][i]-2, 0)
+
+        return self._read_next_timestep()
+
+    def _read_next_timestep(self):
+
+        self._frame += 1
+        if (self._frame >= self.n_frames):
+            raise EOFError('Trying to go over trajectory limit')
+
+        raw_framedata = self._read_GROMOS11_trajectory()
+        self._frame_to_ts(raw_framedata, self.ts)
+
+        return self.ts
+
+    def _reopen(self):
+        """Close and reopen the trajectory"""
+        self.close()
+        self.open_trajectory()
+
+    def open_trajectory(self):
+        if self.trcfile is not None:
+            raise IOError(
+                errno.EALREADY, 'TRC file already opened', self.filename)
+
+        # Reload trajectory file
+        self.trcfile = util.anyopen(self.filename)
+
+        # Reset ts
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+
+        # Set frame to -1, so next timestep is zero
```

Comment:
Do the same in `__init__`

Reply:
I think that the marked line is called in `__init__`

 `__init__` calls `self._reopen()` which then calls `self.open_trajectory()` which then resets `self.ts`

As written in one of the comments above, it's all a bit nested. 
I found the internal API in the beginning hard to work with and each of the other readers does it a little bit differently. If the API should be used differently, I would need a reference implementation of a reader as minimum working example - so that I can rework this reader after that example.

Reply:
Fair enough, the readers are not internally consistent, so fair comment.